### PR TITLE
Create default profile in kustomize

### LINKED ISSUE #3639: Kfctl create profile for user by default
With 0.6.0 a user can no longer create notebooks in the "kubeflow" namespace. They need to create a profile (see https://bit.ly/2UOzzxB) which will create the namespace with appropriate service accounts.

```
apiVersion: kubeflow.org/v1alpha1
kind: Profile
metadata:
  name: <name of the name space to be used>
spec:
  owner:
    kind: User
    name: <ID of the user to use it>
```

We are planning a UI based flow to do that but the UI hasn't landed yet.

In the meantime, we could have `kfctl` automatically create profile for ${USER} running kfctl.
This would mean out of the box the user has a namespace they can run Kubeflow workloads in.

@yanniszark @kkasravi WDYT? Are either of you able to pick this up?

### PR DESCRIPTION
**Pull request by kkasravi**
fixes #3639

Note: v2.zip needed to be updated with a 'v2' version of kubeflow/components/profile-controller.
This means the following files were added to v2.zip where all imports were changed to reference v2 libraries for k8s.io/{api,apimachinery,apiextensions},sigs.k8s.io/controller-runtime and are the following:
```
/tmp/v2/github.com/kubeflow/kubeflow/components/profile-controller/go.mod
/tmp/v2/github.com/kubeflow/kubeflow/components/profile-controller/pkg/apis/istiorbac/v1alpha1/register.go
/tmp/v2/github.com/kubeflow/kubeflow/components/profile-controller/pkg/apis/istiorbac/v1alpha1/servicerolebinding_types.go
/tmp/v2/github.com/kubeflow/kubeflow/components/profile-controller/pkg/apis/istiorbac/v1alpha1/v1alpha1_suite_test.go
/tmp/v2/github.com/kubeflow/kubeflow/components/profile-controller/pkg/apis/istiorbac/v1alpha1/servicerolebinding_types_test.go
/tmp/v2/github.com/kubeflow/kubeflow/components/profile-controller/pkg/apis/istiorbac/v1alpha1/servicerole_types_test.go
/tmp/v2/github.com/kubeflow/kubeflow/components/profile-controller/pkg/apis/istiorbac/v1alpha1/doc.go
/tmp/v2/github.com/kubeflow/kubeflow/components/profile-controller/pkg/apis/istiorbac/v1alpha1/zz_generated.deepcopy.go
/tmp/v2/github.com/kubeflow/kubeflow/components/profile-controller/pkg/apis/istiorbac/v1alpha1/servicerole_types.go
/tmp/v2/github.com/kubeflow/kubeflow/components/profile-controller/pkg/apis/addtoscheme_kubeflow_v1alpha1.go
/tmp/v2/github.com/kubeflow/kubeflow/components/profile-controller/pkg/apis/apis.go
/tmp/v2/github.com/kubeflow/kubeflow/components/profile-controller/pkg/apis/kubeflow/group.go
/tmp/v2/github.com/kubeflow/kubeflow/components/profile-controller/pkg/apis/kubeflow/v1alpha1/register.go
/tmp/v2/github.com/kubeflow/kubeflow/components/profile-controller/pkg/apis/kubeflow/v1alpha1/v1alpha1_suite_test.go
/tmp/v2/github.com/kubeflow/kubeflow/components/profile-controller/pkg/apis/kubeflow/v1alpha1/profile_types_test.go
/tmp/v2/github.com/kubeflow/kubeflow/components/profile-controller/pkg/apis/kubeflow/v1alpha1/profile_types.go
/tmp/v2/github.com/kubeflow/kubeflow/components/profile-controller/pkg/apis/kubeflow/v1alpha1/doc.go
/tmp/v2/github.com/kubeflow/kubeflow/components/profile-controller/pkg/apis/kubeflow/v1alpha1/zz_generated.deepcopy.go
```
Once we remove the need for v2 this can be removed. 

---
This change is [Reviewable](https://reviewable.io/reviews/kubeflow/kubeflow/3673)

### CODE REVIEW

File: bootstrap/v2/pkg/kfapp/kustomize/kustomize.go [hunk 1]
```diff
@@ -227,6 +230,38 @@ func (kustomize *kustomize) Apply(resources kftypes.ResourceEnum) error {
...
+	if kustomize.kfDef.Spec.Email != "" {
+		profile := &profilev2.Profile{
+			TypeMeta: metav1.TypeMeta{
+				Kind:       "Profile",
+				APIVersion: "kubeflow.org/v1alpha1",
+			},
+			ObjectMeta: metav1.ObjectMeta{
+				//TODO this should be unique across namespaces
+				//perhaps Email where '.', '@' are replaced with valid chars
+				Name: "kubeflow-admin",
```
Comment (jlewi) [inline bootstrap/v2/pkg/kfapp/kustomize/kustomize.go]: nit: Can we call this kubeflow-user and not kubeflow-admin?

**Comment by jlewi** [comment]
Test failure

```
INFO|2019-07-16T21:05:03|/src/kubeflow/testing/py/kubeflow/testing/util.py|71| level=error msg="find objects: Received status code '404' when trying to retrieve OpenAPI schema for cluster version 'v0.3.1' from URL 'https://raw.githubusercontent.com/kubernetes/kubernetes/v0.3.1/api/openapi-spec/swagger.json'"
Traceback (most recent call last):
  File "/usr/lib/python2.7/runpy.py", line 174, in _run_module_as_main
    "__main__", fname, loader, pkg_name)
  File "/usr/lib/python2.7/runpy.py", line 72, in _run_code
    exec code in run_globals
  File "/src/kubeflow/testing/py/kubeflow/testing/run_e2e_workflow.py", line 418, in <module>
    final_result = main()
  File "/src/kubeflow/testing/py/kubeflow/testing/run_e2e_workflow.py", line 408, in main
    return run(args, file_handler)
  File "/src/kubeflow/testing/py/kubeflow/testing/run_e2e_workflow.py", line 274, in run
    util.run([ks_cmd, "show", env, "-c", w.component], cwd=w.app_dir)
  File "/src/kubeflow/testing/py/kubeflow/testing/util.py", line 87, in run
    " ".join(command), process.returncode), "\n".join(output))
subprocess.CalledProcessError: Command 'cmd: ks show kubeflow-presubmit-dashboard-release-3673-ace7d4e-2592-2b6e -c centraldashboard exited with code 1' returned non-zero exit status 1
```

SHould be fixed on master

**Comment by gabrielwen** [comment]
submitted with #3676 for rebasing.  closing this.

/close

**Pull request closed without merging**